api/shared/compare/serializers.py

```diff
@@ -76,3 +90,172 @@ class FlagComparisonSerializer(serializers.Serializer):
+    hashed_path = serializers.SerializerMethodField()
+    segments = serializers.SerializerMethodField()
+
+    def get_base_coverage(self, impacted_file: ImpactedFile) -> serializers.JSONField:
+        if impacted_file.base_coverage:
+            return dataclasses.asdict(impacted_file.base_coverage)
+
+    def get_head_coverage(self, impacted_file: ImpactedFile) -> serializers.JSONField:
+        if impacted_file.head_coverage:
+            return dataclasses.asdict(impacted_file.head_coverage)
+
+    def get_patch_coverage(self, impacted_file: ImpactedFile) -> serializers.JSONField:
+        if impacted_file.patch_coverage:
+            return dataclasses.asdict(impacted_file.patch_coverage)
+
+    def get_file_name(self, impacted_file: ImpactedFile) -> serializers.CharField:
+        return impacted_file.file_name
+
+    def get_is_new_file(self, impacted_file: ImpactedFile) -> serializers.BooleanField:
+        base_name = impacted_file.base_name
+        head_name = impacted_file.head_name
+        return base_name is None and head_name is not None
+
+    def get_is_renamed_file(
+        self, impacted_file: ImpactedFile
+    ) -> serializers.BooleanField:
+        base_name = impacted_file.base_name
+        head_name = impacted_file.head_name
+        return (
+            base_name is not None and head_name is not None and base_name != head_name
+        )
+
+    def get_is_deleted_file(
+        self, impacted_file: ImpactedFile
+    ) -> serializers.BooleanField:
+        base_name = impacted_file.base_name
+        head_name = impacted_file.head_name
+        return base_name is not None and head_name is None
+
+    def get_change_coverage(
+        self, impacted_file: ImpactedFile
+    ) -> serializers.FloatField:
+        return impacted_file.change_coverage
+
+    def get_misses_count(self, impacted_file: ImpactedFile) -> serializers.IntegerField:
+        return impacted_file.misses_count
+
+    def get_hashed_path(self, impacted_file: ImpactedFile) -> serializers.CharField:
+        path = impacted_file.head_name
+        encoded_path = path.encode()
+        md5_path = hashlib.md5(encoded_path)
+        return md5_path.hexdigest()
+
+    def get_segments(
+        self, impacted_file: ImpactedFile
+    ) -> ImpactedFileSegmentSerializer:
+        file_comparison = self.context["comparison"].get_file_comparison(
+            impacted_file.head_name, with_src=True, bypass_max_diff=True
+        )
+        return [
+            ImpactedFileSegmentSerializer(segment).data
+            for segment in file_comparison.segments
+        ]
+
+
+class ImpactedFilesComparisonSerializer(ComparisonSerializer):
+
+    files = serializers.SerializerMethodField()
+
+    def get_files(self, comparison: Comparison) -> List[dict]:
+        comparison_table = CommitComparison._meta.db_table
+        commit_table = Commit._meta.db_table
+        commit_comparison = CommitComparison.objects.raw(
```

Comment:
Since this is a sort of complicated raw SQL query and could be error prone if updated maybe we pull this out somewhere and reuse it both in the dataloader and here.

Reply:
Moved to `services.CommitComparisonService`, both public and GQL APIs are getting the query from there.